packages/slice-machine/src/modules/slices/index.ts

```diff
@@ -91,11 +136,22 @@ export function* createSliceSaga({
     name: payload.sliceName,
     library: payload.libName,
   });
+  const { data: serverState } = (yield call(getState)) as SagaReturnType<
+    typeof getState
+  >;
+  yield put(createSliceCreator.success({ libraries: serverState.libraries }));
   yield put(modalCloseCreator({ modalKey: ModalKeysEnum.CREATE_SLICE }));
-  // changing this to use next/router  call(Router.router.push,...) causes bugs in some contexts.
-  window.location.href = `/${payload.libName.replace(/\//g, "--")}/${
+  const addr = `/${payload.libName.replace(/\//g, "--")}/${
     payload.sliceName
   }/${variationId}`;
+  yield put(replace("/[lib]/[sliceName]/[variation]", addr));
```

Comment:
is it replace or put, or are the two similar?

Reply:
good point it's `push`. The difference between replace and push is that push adds a new route into the history route stack, while replace changes the top one. Therefore we want to use `push` so that we add on the stack and we can navigate back. 

A usecase of `replace` could be if the user tries to go to a wrong URL, in which case we want to override it so that he cannot navigate back to the wrong route. 

https://stackoverflow.com/questions/52954837/difference-between-react-router-push-and-router-replace#:~:text=replace%20%2C%20you're%20overwritting%20the,back%20to%20the%20last%20page.